modules/core/src/main/java/org/apache/ignite/internal/processors/cache/verify/VerifyBackupPartitionsTaskV2.java

```diff
@@ -259,6 +271,51 @@ else if (e.getCause() instanceof IgniteException)
             return res;
         }
 
+        /**
+         * @return True if validates only specific kind caches, else false.
+         */
+        private boolean onlySpecificKindCaches() {
+            if (arg instanceof VisorIdleVerifyDumpTaskArg) {
+                VisorIdleVerifyDumpTaskArg vdta = (VisorIdleVerifyDumpTaskArg)arg;
+
+                if (vdta.getCacheFilterEnum() != CacheFilterEnum.ALL)
+                    return true;
+            }
+
+            return false;
+        }
+
+        /**
+         * @param cacheName Cache name.
+         */
+        private boolean isCacheMatchKind(String cacheName) {
+            if (arg instanceof VisorIdleVerifyDumpTaskArg) {
+                DataStorageConfiguration dsc = ignite.context().config().getDataStorageConfiguration();
+                DynamicCacheDescriptor desc = ignite.context().cache().cacheDescriptor(cacheName);
+                CacheConfiguration cc = desc.cacheConfiguration();
+                VisorIdleVerifyDumpTaskArg vdta = (VisorIdleVerifyDumpTaskArg)arg;
+
+                switch (vdta.getCacheFilterEnum()) {
```

Comment:
It seems that `VisorIdleVerifyDumpTaskArg.getCacheKind()` can return `null` value in case that the task is triggered by a node with a previous version (V1) of that task.

Reply:
I changed readExternalData method in order to that cannot be null.